### PR TITLE
PLAT-72718: Scrollable and VirtualList: Update lifecycle method

### PR DESCRIPTION
**Pull request by 0x64**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Removed deprecated _componentWillReceiveProps_ methods used to configure Spotlight containers.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Replaced _componentWillReceiveProps_ with _componentDidUpdate_ method.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
There was a missing code to reduce the number of calls of _Spotlight.set_ in Scrollable, so I fixed it also.

### Links
[//]: # (Related issues, references)
PLAT-72718

### CODE REVIEW

File: packages/moonstone/VirtualList/VirtualListBase.js [hunk 1]
```diff
@@ -264,6 +258,7 @@ const VirtualListBaseFactory = (type) => {
 		nodeIndexToBeFocused = null
 		preservedIndex = null
 		restoreLastFocused = false
+		lastConfiguredSpotlightId = null
```
Comment (ybsung) [inline packages/moonstone/VirtualList/VirtualListBase.js]: How about `prevSpotlightId` as you did in `Scrollable` instead of `lastConfiguredSpotlightId`?

**Comment by ybsung** [comment]
LGTM

File: packages/moonstone/Scrollable/Scrollable.js [hunk 2]
```diff
@@ -236,10 +236,28 @@ class ScrollableBase extends Component {
...
 
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		configureSpotlightContainer(nextProps);
+	static getDerivedStateFromProps (props, state) {
+		const
+			{'data-spotlight-id': spotlightId, focusableScrollbar} = props,
+			{prevSpotlightId, prevFocusableScrollbar} = state;
+
+		if (prevSpotlightId !== spotlightId || prevFocusableScrollbar !== focusableScrollbar) {
+			configureSpotlightContainer(props);
```
Comment (webOS101) [inline packages/moonstone/Scrollable/Scrollable.js]: We really, really shouldn't be doing side effects within this static method.  Also, there should be a better way to deal with this other than adding state just for the sole purpose of having it available within `getDerivedStateFromProps`.

**Comment by 0x64** [comment]
I updated to remove side effects from the static method.

**Comment by 0x64** [comment]
The call in constructor looks good for now. To reduce calls to Spotlight, I left code to check the modification of props and the initial Spotlight configuring is needed at least once at the beginning.

**Comment by 0x64** [comment]
> Any reason why we can't run `configureSpotlightContainer` in `componentDidUpdate`?

`render` is executed in _render phase_ and `componentDidUpdate` is in _commit phase_, so I just wanted to set up spotlight config in _render phase_ since a focus could be moved out of a scrollable area if a user input occurs during updates. But, I think now that I don't need to consider such an abnormal scenario which cannot happen in apps. So, it looks better to move the code. I will update. Thanks.

**Comment by 0x64** [comment]
Updated.

**Comment by viodragon2** [comment]
I was thinking more like this https://github.com/enactjs/enact/compare/develop...feature/PLAT-72718-stephen

**Comment by 0x64** [comment]
Oops, what a silly mistake I did! Thanks.

**Pull request merged**